modules/service-fips.sh

```diff
@@ -50,6 +54,62 @@ fips_disable() {
     not_supported 'Disabling FIPS'
 }
 
+fips_update() {
+    local token="$1"
+    local bypass_prompt="$2"
+    local fips_configured=0
+
+    check_token "$FIPS_UPDATES_REPO_URL" "$token"
+
+    echo -n "Updating FIPS packages will take the system out of FIPS compliance."
+    if [ "$bypass_prompt" -ne 1 ]; then
+        if ! prompt_user 'Do you want to proceed?'; then
+            error_msg "Aborting updating FIPS packages..."
+            return
+        fi
+    fi
+
+    # add the fips-updates repo if the system is undergoing updates the first time
+    if [ ! -f "$FIPS_UPDATES_REPO_LIST" ]; then
+        apt_add_repo "$FIPS_UPDATES_REPO_LIST" "$FIPS_UPDATES_REPO_URL" "$token" \
+                 "${KEYRINGS_DIR}/${FIPS_UPDATES_REPO_KEY_FILE}"
+        apt_add_repo_pinning "$FIPS_UPDATES_REPO_PREFERENCES" \
+                         LP-PPA-ubuntu-advantage-fips-updates 1001
+        apt_install_package_if_missing_file "$APT_METHOD_HTTPS" apt-transport-https
+        apt_install_package_if_missing_file "$CA_CERTIFICATES" ca-certificates
+        echo -n 'Running apt-get update... '
+        check_result apt_get update
+        echo 'Ubuntu FIPS-UPDATES PPA repository enabled.'
+    fi
```

Comment:
Do we want to keep going if the updates repo is already there? What is the user story here where I would run `ua update-fips <token>` multiple times?

Reply:
I think we would need to do updates via the ua tool itself as 
a) kernel update will need update-grub to be run
b) Not all customers may want updates.

This brings me to another question. I am open to suggestions here. Currently the implementation does not check if the packages installed are latest. Invoking the update switch multiple times just tries to update all packages and asks to reboot. I am thinking we may need to add some apt checking (apt-cache policy <pkg> or apt list -a --upgradable) to make sure if all installed packages are latest and just return if so. Ask to reboot only if kernel has been updated.

Reply:
`ua update-fips <token>` shouldn't be needed anymore after it was enabled. Package updates, kernel included, should be handled by the normal ubuntu upgrade process for deb packages. Unless the fips kernel package does or requires something different.

Which updates to install, and how, is always a customer's choice. Or are you talking about `unattended-upgrades`? AFAIK that is enabled by default for the security pocket.

Reply:
Agree. After offline discussion via email on this, I understand now that ua update-fips <token> is meant to be used only once to enable fips-updates repo and install updates. The future updates are going to be installed via apt upgrade.

Reply:
I did see ESM added to unattended-upgrades ( /etc/apt/apt.conf.d/50unattended-upgrades) and it installs where applicable in addition to the security pocket. 
For FIPS, I prefer leaving it to customer which updates to install as certification is involved.

Reply:
Ok, so what's the end result here? `enable-fips-updates` should check if `fips-updates` is already enabled? Right now it's not doing that, maybe this change was forgotten, or I misunderstood it.

Right now `fips_updates_enable()` does, in this order:
 * checks the token (network call)
 * optionally prompts the user for confirmation (bypassed with `-y`)
 * checks if the updates repo file exists and, if not, adds it and installs needed dependencies
 * installs (new) fips packages. Will be noop if the updates were already installed
 * configures fips (bootloader, etc) if this is the first time fips is installed, i.e., the user isn't coming from `enable-fips` but is going straight to `enable-fips-updates`

I was expecting that, if `fips-updates` is already enabled, all the steps above would be skipped.

Reply:
Earlier this function was understood to be used every time an update was to be installed. It should skip all the steps if fips-updates is already configured manually or via tool. I will add in that check before user prompt.